configure

```diff
@@ -106,6 +107,10 @@ dragonfly*|freebsd*)
 	if [ -z "$STATUSARG" ]; then
 		STATUSARG="onestatus"
 	fi
+	echo "STAT_FMT=         stat -f" >>$CONFIG_MK
+	;;
+*bsd*)
```

Comment:
I think the whole stat check needs to be in it's own section.
I'm tempted to suggest that we assume BSD style by default and add an exception for Linux.

The only potential issue I think is kFreeBSD which may use the coreutils stat rather than the FreeBSD one.
Also some stat's just don't offer the functionality we need such as Plan9 or disabled as in the case for busybox.

Reply:
Those unusual cases should be able to handle by ``make STAT_FMT=``. I am not sure how often is crossbuilding done for such systems. I think configure could test the output by actually checking ``stat -c %d .`` and that it does report  a number. Not sure if such check would work on busybox systems. At least stat from alpine docker image and BusyBox v1.36.1 has -c %d support.

I expect Linux systems are dominant today. But if the BSD variant is likely to work on different systems, which may use openresolv, it would be okay. It should have own case "$OS" just for stat check, is that what you mean?